### DISCUSSION TITLE
Webhook URL

### DISCUSSION THREAD

**Discussion by FehmiBB** [netbox-community/netbox#6458] [Q&A]
Hi all,
I would like to add the name of a VM in the URL of a Webhook. like: https://example.com/{VM.Name}

Does NetBox support this kind of implementation?

**Comment by candlerb** [comment]
There is no customization of either the target URL or the payload format.  You would need to write your own external middleware, which receives a webhook message, and then triggers a new request with a new URL and/or payload.

**Reply by jeremystretch** [reply]
That's not quite accurate anymore. NetBox v2.7.8 introduced support for Jinja2 templating of the webhook payload (#4237). The HTTP headers can also be templatized. The destination URL, however, cannot.

I'd be curious to hear the use case for it.

**Reply by romanstech** [reply]
Jeremy, I use Jinja2 template extensively in Custom Links section and sometimes 500 characters limit is not enough when I use conditionals. Do you have any real reason for such limit or it's possible to increase it?
